provider/cloudflare/cloudflare.go

```diff
@@ -457,19 +457,18 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 					failedChange = true
 					log.WithFields(logFields).Errorf("failed to delete record: %v", err)
 				}
-				if change.CustomHostname.Hostname == "" {
-					continue
-				}
-				log.WithFields(logFields).Infof("Deleting custom hostname %v", change.CustomHostname.Hostname)
-				chID, _ := p.getCustomHostnameOrigin(chs, change.CustomHostname.Hostname)
-				if chID == "" {
-					log.WithFields(logFields).Infof("Custom hostname %v not found", change.CustomHostname.Hostname)
-					continue
-				}
-				chErr := p.Client.DeleteCustomHostname(ctx, zoneID, chID)
-				if chErr != nil {
-					failedChange = true
-					log.WithFields(logFields).Errorf("failed to delete custom hostname %v/%v: %v", chID, change.CustomHostname.Hostname, chErr)
+				if recordTypeCustomHostnameSupported[change.ResourceRecord.Type] && change.CustomHostname.Hostname != "" {
+					log.WithFields(logFields).Infof("Deleting custom hostname %v", change.CustomHostname.Hostname)
+					chID, _ := getCustomHostnameIdWithOrigin(chs, change.CustomHostname.Hostname)
+					if chID != "" {
+						chErr := p.Client.DeleteCustomHostname(ctx, zoneID, chID)
+						if chErr != nil {
+							failedChange = true
+							log.WithFields(logFields).Errorf("failed to delete custom hostname %v/%v: %v", chID, change.CustomHostname.Hostname, chErr)
+						}
+					} else {
+						log.WithFields(logFields).Infof("Custom hostname %v not found", change.CustomHostname.Hostname)
```

Comment:
```suggestion
						log.WithFields(logFields).Warnf("Custom hostname %v not found", change.CustomHostname.Hostname)
```

Reply:
%v -> %q in all kind of log message please!

Reply:
> %v -> %q in all kind of log message please!

Updated for Cloudflare provider.
There's still a logging inconsistency with the logrus's `log.WithFields()`, looking like:
```
INFO[0001] Changing record.                              action=CREATE record=mih-cf-test-2.***.xyz ttl=1 type=A zone=***
INFO[0001] Creating custom hostname "mih-test-ch-1.***"  action=CREATE record=mih-cf-test-2.*** ttl=1 type=A zone=***
INFO[0002] Changing record.                              action=CREATE record=a-mih-cf-test-2.*** ttl=1 type=TXT zone=***
```
If the consistent quoted output is a style policy goal, we could use `log.SetFormatter(&log.TextFormatter{ForceQuote: true})`.

Should we go for it?